pkg/sdn/plugin/cniserver.go

```diff
@@ -0,0 +1,213 @@
+package plugin
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"os"
+	"path"
+	"strings"
+
+	"github.com/golang/glog"
+	"github.com/gorilla/mux"
+)
+
+const CniServerSocketPath string = "/var/run/openshift-sdn/cni-server.sock"
+
+type CniCommand string
+
+const CNI_ADD CniCommand = "ADD"
+const CNI_UPDATE CniCommand = "UPDATE"
+const CNI_DEL CniCommand = "DEL"
+
+type CniRequest struct {
+	// CNI environment variables, like CNI_COMMAND and CNI_NETNS
+	Env map[string]string `json:"env,omitempty"`
+	// CNI configuration passed via stdin to the CNI plugin
+	Config []byte `json:"config,omitempty"`
+}
+
+type podRequest struct {
+	command   CniCommand
+	namespace string
+	name      string
+	id        string
+	netns     string
```

Comment:
I think all usage of 'namespace' in the IPAM commit is actually the namespace name.  The stuff that's kernel-netns related is called "netns".